src/Component/AbstractComponent.php

```diff
@@ -275,16 +275,30 @@ final public function toManagementApiData () : array
 
 		$definition = $this->configureComponent();
 		$previewField = null;
+		$previewImageField = null;
 
 		foreach ($this->getFields()->getRootFields() as $key => $field)
 		{
 			if ($field->isStoryblokPreviewField())
 			{
 				$previewField = $key;
-				break;
+
+				if (null !== $previewField && null !== $previewImageField)
+				{
+					break;
+				}
 			}
-		}
 
+			if ($field->isStoryblokImagePreviewField())
+			{
+				$previewImageField = $key;
+
+				if (null !== $previewField && null !== $previewImageField)
```

Comment:
Put this check at the bottom.

Or maybe add a check if you set multiple once, which would also be a bug.